From 9720c17724c86c0c065bb5ad3bfc892b3a07d956 Mon Sep 17 00:00:00 2001
From: Louis Marelle <louis.marelle@latmos.ipsl.fr>
Date: Mon, 28 Feb 2022 17:16:15 +0100
Subject: [PATCH 17/55] Update MOSAIC aerosol sumpm

In chem/module_mosaic_sumpm.F, remove the VBS species from the
calculation of PM for MOSAIC mechanisms without VBS.
---
 chem/module_mosaic_sumpm.F | 156 +------------------------------------
 1 file changed, 2 insertions(+), 154 deletions(-)

diff --git a/chem/module_mosaic_sumpm.F b/chem/module_mosaic_sumpm.F
index c35cc0b9..441f5ad4 100644
--- a/chem/module_mosaic_sumpm.F
+++ b/chem/module_mosaic_sumpm.F
@@ -169,85 +169,7 @@
                        + chem(i,k,j,lptr_na_aer(n,itype,iphase))  &
                        + chem(i,k,j,lptr_oin_aer(n,itype,iphase)) &
                        + chem(i,k,j,lptr_oc_aer(n,itype,iphase))  &
-                       + chem(i,k,j,lptr_bc_aer(n,itype,iphase))  &
-                       + chem(i,k,j,lptr_pcg1_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg2_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg3_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg4_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg5_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg6_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg7_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg1_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg2_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg3_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg4_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg5_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg6_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg7_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg1_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg2_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg3_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg4_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg5_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg6_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg7_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg1_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg2_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg3_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg4_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg5_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg6_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg7_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg1_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg2_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg3_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg4_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg5_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg6_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg7_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg1_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg2_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg3_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg4_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg5_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg6_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg7_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg1_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg2_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg3_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg4_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg5_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg6_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg7_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg1_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg2_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg3_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg4_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg5_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg6_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg7_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_ant1_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant2_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant3_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant4_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant1_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant2_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant3_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant4_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog1_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog2_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog3_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog4_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog1_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog2_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog3_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog4_o_aer(n,itype,iphase))    
-
-
-
-
- 
-
+                       + chem(i,k,j,lptr_bc_aer(n,itype,iphase))
  
                   pm2_5_dry(i,k,j) = pm2_5_dry(i,k,j) + mass
 
@@ -273,81 +195,7 @@
                               + chem(i,k,j,lptr_na_aer(n,itype,iphase))  &
                               + chem(i,k,j,lptr_oin_aer(n,itype,iphase)) &
                               + chem(i,k,j,lptr_oc_aer(n,itype,iphase))  &
-                              + chem(i,k,j,lptr_bc_aer(n,itype,iphase))  &
-                       + chem(i,k,j,lptr_pcg1_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg2_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg3_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg4_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg5_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg6_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg7_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg1_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg2_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg3_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg4_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg5_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg6_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg7_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg1_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg2_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg3_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg4_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg5_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg6_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg7_b_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg1_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg2_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg3_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg4_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg5_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg6_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg7_b_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg1_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg2_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg3_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg4_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg5_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg6_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg7_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg1_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg2_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg3_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg4_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg5_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg6_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_pcg7_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg1_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg2_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg3_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg4_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg5_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg6_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg7_f_c_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg1_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg2_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg3_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg4_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg5_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg6_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_opcg7_f_o_aer(n,itype,iphase)) &
-                       + chem(i,k,j,lptr_ant1_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant2_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant3_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant4_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant1_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant2_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant3_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_ant4_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog1_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog2_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog3_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog4_c_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog1_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog2_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog3_o_aer(n,itype,iphase))    &
-                       + chem(i,k,j,lptr_biog4_o_aer(n,itype,iphase))
-
-
+                              + chem(i,k,j,lptr_bc_aer(n,itype,iphase))
 
                enddo
             enddo
-- 
2.31.1

